modal/_utils/async_utils.py

```diff
@@ -217,13 +217,20 @@ async def wait(self, *tasks):
                 )
             except asyncio.TimeoutError:
                 continue
+            except ValueError:
+                # The `add_done_callback` of the task races with the `wait`, raising a ValueError
+                # because `self._tasks` is empty.
+                break
             for task in done:
                 task.result()  # Raise exception if needed
                 if task in unfinished_tasks:
                     unfinished_tasks.remove(task)
                 if task in self._tasks:
                     self._tasks.remove(task)
 
+    async def wait_all(self):
+        await self.wait(*self._tasks)
```

Comment:
I don't think `TaskContext.wait_all()` makes sense as a concurrency primitive. Can we adjust your code here to use `asyncio.gather()` instead on the returned tasks?